Sources/PackageModel/Target.swift

```diff
@@ -638,6 +638,11 @@ public final class BinaryTarget: Target {
         }
     }
 
+    public var containsExecutable: Bool {
+        // FIXME: needs to be revisited once libraries are supported in artifact bundles
```

Comment:
I opted to not open this can of worms here since right now `.artifactsArchive` implies that there is an executable. Addressing this required too many changes that are completely unrelated to this PR and which should be done once we implement support for libraries in artifact bundles.

Reply:
I think this is fair, but would be nice if there was some kind of associated data on the artifactsArchive that would allow us to throw an error if it is not an executable. eg `.artifactsArchive(type:)` not sure if that is the can of warms tho :)

Reply:
Yah, that was my approach, but it did require a bunch of changes 🪱  Could be a good change for a separate PR, though.